subsys/bluetooth/audio/stream.c

```diff
@@ -859,22 +830,48 @@ static void unicast_client_codec_qos_to_iso_qos(struct bt_audio_iso *iso,
 	}
 }
 
-static int unicast_group_add_stream(struct bt_audio_unicast_group *group,
-				    struct bt_audio_stream *stream,
-				    struct bt_codec_qos *qos,
-				    enum bt_audio_dir dir)
+static void unicast_group_add_stream(struct bt_audio_unicast_group *group,
+				     struct bt_audio_stream *stream,
+				     struct bt_codec_qos *qos,
+				     struct bt_audio_iso *iso,
+				     enum bt_audio_dir dir)
+{
+	LOG_DBG("group %p stream %p qos %p iso %p dir %d",
+		group, stream, qos, iso, dir);
+
+	__ASSERT_NO_MSG(stream->ep == NULL ||
+			(stream->ep != NULL && stream->ep->iso == NULL));
+
+	stream->qos = qos;
+	stream->dir = dir;
+	stream->unicast_group = group;
+
+	/* iso initialized already */
+	bt_audio_iso_bind_stream(iso, stream);
+	if (stream->ep != NULL) {
+		bt_audio_iso_bind_ep(iso, stream->ep);
+	}
+
+	/* Store the Codec QoS in the audio_iso */
+	unicast_client_codec_qos_to_iso_qos(iso, qos, dir);
+
+	sys_slist_append(&group->streams, &stream->_node);
+}
+
+static int unicast_group_add_stream_pair(struct bt_audio_unicast_group *group,
+					 struct bt_audio_stream *in,
+					 struct bt_audio_stream *out,
+					 struct bt_codec_qos *qos)
```

Comment:
Given that the last 3 arguments are always in the `bt_audio_unicast_group_stream_param`, we could just do
```suggestion
static int unicast_group_add_stream_pair(struct bt_audio_unicast_group *group,
					 struct bt_audio_unicast_group_stream_param *param)
```